src/test/java/org/openlmis/stockmanagement/service/StockCardAggregateTest.java

```diff
@@ -221,31 +231,42 @@ public void shouldGetStockOutDaysInRange() {
     assertEquals(new Long(0), stockCardAggregate.getStockoutDays(
         LocalDate.of(2018, 5, 12), LocalDate.of(2018, 5, 13)));
 
-    assertEquals(new Long(0), stockCardAggregate.getStockoutDays(
+    assertEquals(new Long(16), stockCardAggregate.getStockoutDays(
```

Comment:
i'm not sure about this expected value, because there is only one row with stockCard4 in 15-06-2018 with SOH 0, so I think for June it should be 29 expected

Reply:
Calculations of the **getStockoutDays** method are based on the list of [CalculatedStockOnHand](https://github.com/OpenLMIS/openlmis-stockmanagement/blob/master/src/main/java/org/openlmis/stockmanagement/domain/event/CalculatedStockOnHand.java) objects that will be passed to the class in which it is located.
Therefore, despite assigning different stock cards to these objects, they are taken into account together in the calculations (getStockoutDays pays attention only to the date of occurrence and the amount of stock on hand, not to stock cards).
In the OLMIS, the [StockCardRangeSummaryBuilder](https://github.com/OpenLMIS/openlmis-stockmanagement/blob/master/src/main/java/org/openlmis/stockmanagement/web/stockcardrangesummary/StockCardRangeSummaryBuilder.java) class that uses the getStockoutDays method takes care of passing only objects related to a specific product to the class of this method.

Reply:
ok, makes sense